src/components/Notifications/EventLog/usePrimaryToolbarFilterConfigWrapper.tsx

```diff
@@ -0,0 +1,230 @@
+import { ColumnsMetada, usePrimaryToolbarFilterConfig } from '@redhat-cloud-services/insights-common-typescript';
+import produce from 'immer';
+import React from 'react';
+
+import { Schemas } from '../../../generated/OpenapiNotifications';
+import { areEqual } from '../../../utils/Arrays';
+import { ClearEventLogFilters, EventLogFilterColumn, EventLogFilters, SetEventLogFilters } from './EventLogFilter';
+import { EventLogTreeFilter } from './EventLogTreeFilter';
+
+export interface EventLogCustomFilter {
+    bundleId: string,
+    category: string,
+    chips: Array<{ name: string, value: string, isRead: boolean }>
+}
+
+// Wrapper hook that gets the PrimaryToolbarFilterConfig and adds a custom conditional filter using Dropdown/Tree components
+// usePrimaryToolbarFilterConfig only supports 3 filter types: checkbox, radio, and text, so this extends that
+export const usePrimaryToolbarFilterConfigWrapper = (
+    bundles: readonly Schemas.Facet[],
+    filters: EventLogFilters,
+    setFilters: SetEventLogFilters,
+    clearFilter: ClearEventLogFilters,
+    metaData: ColumnsMetada<typeof EventLogFilterColumn>
+) => {
+    const [ customFilters, setCustomFilters ] = React.useState([] as EventLogCustomFilter[]);
+    const toolbarConfig = usePrimaryToolbarFilterConfig(
+        EventLogFilterColumn,
+        filters,
+        setFilters,
+        clearFilter,
+        metaData
+    );
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const defaultDelete = React.useMemo(() => toolbarConfig.activeFiltersConfig.onDelete, []);
+    toolbarConfig.activeFiltersConfig.onDelete = React.useCallback((
```

Comment:
I didn't notice this before, but we should create a new object instead of modifying what the previous hook gave us.

Reply:
would wrapping everything in `produce(toolbarConfig, () =>{...})` be the best solution for that or simply using the `...` spread operator?

Reply:
a produce would be good. We can put all our updates in variables and modify at the end.